atest/interpreter.py

```diff
@@ -15,16 +15,16 @@ def get_variables(path, name=None, version=None):
     return {'INTERPRETER': interpreter, 'UNICODE PREFIX': u}
 
 
-def InterpreterFactory(path, name=None, version=None):
+def InterpreterFactory(path, name=None, version=None, coverage=False):
     if path.endswith('.jar'):
-        return StandaloneInterpreter(path, name, version)
-    return Interpreter(path, name, version)
+        return StandaloneInterpreter(path, name, version, coverage)
+    return Interpreter(path, name, version, coverage)
 
 
 class Interpreter(object):
 
-    def __init__(self, path, name=None, version=None):
-        self.path = path
+    def __init__(self, path, name=None, version=None, coverage=False):
+        self.path = self._get_interpreter_path(path, coverage)
         self.interpreter = self._get_interpreter(path)
```

Comment:
Is it on purpose that `self.path` gets coverage configuration but `self.interpreter` doesn't? I don't remember how all this works together, but I thought `self.interpreter` is what is actually used when running tests.